ablySpec/RestClient.swift

```diff
@@ -45,6 +45,32 @@ class RestClient: QuickSpec {
                     expect(publishTask.error?.code).toEventually(equal(40005))
                 }
 
+                it("should accept a token") {
+                    let options = AblyTests.commonAppSetup()
+                    let client = ARTRest(key: options.key!)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.auth.requestToken(ARTAuthTokenParams(), withOptions: nil) { tokenDetails, error in
+                            if let e = error {
+                                XCTFail(e.description)
+                                done()
+                                return
+                            }
+
+                            guard let tokenDetails = tokenDetails else {
+                                XCTFail("expected tokenDetails not to be nil when error is nil")
+                                done()
+                                return
+                            }
+
+                            let client = ARTRest(token: tokenDetails.token)
```

Comment:
The conversion removes the `initWith`: 

> Objective-C initializers begin with init, or initWith: if the initializer takes one or more arguments. When an Objective-C initializer is imported by Swift, the init prefix becomes an init keyword to indicate that the method is a Swift initializer. If the initializer takes an argument, the With is removed and the rest of the selector is divided up into named parameters accordingly.
https://developer.apple.com/library/ios/documentation/Swift/Conceptual/BuildingCocoaApps/InteractingWithObjective-CAPIs.html#//apple_ref/doc/uid/TP40014216-CH4-ID35